Packet++/header/GvcpLayer.h

```diff
@@ -0,0 +1,758 @@
+		Busy = 0x8007,  ///< A required resource to service the request is not currently available. The request may be
+		                ///< retried at a later time
+		LocalProblem = 0x8008,       // deprecated
+		MsgMismatch = 0x8009,        // deprecated
+		InvalidProtocol = 0x800A,    // deprecated
+		NoMsg = 0x800B,              // deprecated
+		PacketUnavailable = 0x800C,  ///< The requested packet is not available anymore
+		DataOverrun = 0x800D,        ///< Internal memory of GVSP transmitter overrun (typically for image acquisition)
+		InvalidHeader = 0x800E,  ///< The message header is not valid. Some of its fields do not match the specification
+		WrongConfig = 0x800F,    // deprecated
+		PacketNotYetAvailable = 0x8010,  ///< The requested packet has not yet been acquired. Can be used for linescan
+		                                 ///< cameras device when line trigger rate is slower than application timeout
+		PacketAndPrevRemovedFromMemory = 0x8011,  ///< The requested packet and all previous ones are not available
+		                                          ///< anymore and have been discarded from the GVSP transmitter memory
+		PacketRemovedFromMemory = 0x8012,  ///< The requested packet is not available anymore and has been discarded
+		                                   ///< from the GVSP transmitter memory
+		NoRefTime = 0x8013,  ///< The device is not synchronized to a master clock to be used as time reference
+		PacketTemporarilyUnavailable = 0x8014,  ///< The packet cannot be resent at the moment due to temporary
+		                                        ///< bandwidth issues and should be requested again in the future
+		Overflow = 0x8015,                      ///< A device queue or packet data has overflowed
+		ActionLate = 0x8016,  ///< The requested scheduled action command was requested at a time that is already past
+		LeaderTrailerOverflow = 0x8017,  // GEV 2.1
+		Error = 0x8FFF,                  ///< Generic error
+		Unknown = 0xFFFF                 ///< Unknown status
+	};
+
+	std::ostream& operator<<(std::ostream& os, GvcpResponseStatus status);
+
+#pragma pack(push, 1)
+	/// @brief Gvcp request header
+	/// @note refer to the spec "15.1 Request Header". The data is stored as big-endian.
+	struct gvcp_request_header
+	{
+		uint8_t magicNumber = internal::kGvcpMagicNumber;  // always fixed
+		uint8_t flag = 0;  // 0-3 bits are specified by each command, 4-6 bits are reserved, 7 bit is acknowledge
+		uint16_t command = 0;
+		uint16_t dataSize = 0;
+		uint16_t requestId = 0;
+
+		// ------------- methods --------------
+		gvcp_request_header() = default;
+
+		gvcp_request_header(GvcpFlag flag, GvcpCommand command, uint16_t dataSize, uint16_t requestId)
+		    : flag(flag), command(hostToNet16(static_cast<uint16_t>(command))), dataSize(hostToNet16(dataSize)),
+		      requestId(hostToNet16(requestId))
+		{}
+
+		GvcpCommand getCommand() const
```

Comment:
If we decide to keep this struct and the other structs public, we should add doxygen documentation for each public member and public method

Reply:
the struct is now under `internal`.